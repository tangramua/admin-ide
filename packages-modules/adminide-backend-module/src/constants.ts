export const DEFAULT_INTERVAL = 86400000;

export const WORKSPACE_RESOURCE_QUERY = (namespace) => encodeURIComponent(`
sum(
    max(kube_pod_labels{label_release="idestack"}) by (label_release, pod)
    *
    on (pod)
    group_right(label_release)
    label_replace(
        sum by(pod_name) (
            rate(container_cpu_usage_seconds_total{namespace="${namespace}"}[2w])
        ), "pod" , "$1" , "pod_name" , "(.+)"
    )
) by (label_release)
`);

export enum Types {
    BillingWatcher = 'BillingWatcher',
    WorkspaceWatcher = 'WorkspaceWatcher',
    ResourcesService = 'ResourcesService',
    PrometheusService = 'PrometheusService',
}

export enum HemeraTopics {
    Prometheus = 'prometheus',
    Utilization = 'resource-utilization',
}

export enum HemeraCmd {
    Scan = 'scan',
    Query = 'query',
    Collect = 'collect',
}
