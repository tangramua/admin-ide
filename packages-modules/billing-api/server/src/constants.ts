import { IMachineConfigRow } from './interfaces';

export enum EGoogleConstants {
    GCEService = '6F81-5844-456A',
}

export enum EGoogleResourceGroup {
    CPU = 'CPU',
    GPU = 'GPU',
    RAM = 'RAM',
    N1Highmem = 'N1Highmem',
    N1Highcpu = 'N1Highcpu',
    N1Standard = 'N1Standard',
}

export enum EGoogleResourceFamily {
    Network = 'Network',
    License = 'License',
    Compute = 'Compute',
    Storage = 'Storage',
}

export enum HemeraTopics {
    Billing = 'Billing',
}

export enum HemeraCommands {
    Charge = 'charge',
}

export enum ETypes {
    BillingService = 'BillingService',
    PaymentService = 'PaymentService',
    PricingService = 'PricingService',
    BillingReportService = 'BillingReportService',

    PricingApi = 'PricingApi',
    PaymentGateway = 'PaymentGateway',

    PricesRepository = 'PricesRepository',
    BillingReportsRepository = 'BillingReportsRepository',
}

export interface IMachinesValues {
    [propName: string]: IMachineConfigRow;
}

export const MachineConfig: IMachinesValues = {
    HCpu2: { id: 'n1-highcpu-2', cpu: 2, ram: 1800 },
    HCpu4: { id: 'n1-highcpu-4', cpu: 4, ram: 3600 },
    HCpu8: { id: 'n1-highcpu-8', cpu: 8, ram: 7200 },
    HCpu16: { id: 'n1-highcpu-16', cpu: 16, ram: 14400 },
    HCpu32: { id: 'n1-highcpu-32', cpu: 32, ram: 28800 },

    HRam2: { id: 'n1-highmem-2', cpu: 2, ram: 13000 },
    HRam4: { id: 'n1-highmem-4', cpu: 4, ram: 26000 },
    HRam8: { id: 'n1-highmem-8', cpu: 8, ram: 52000 },
    HRam16: { id: 'n1-highmem-16', cpu: 16, ram: 104000 },
    HRam32: { id: 'n1-highmem-32', cpu: 32, ram: 208000 },

    MStandard1: { id: 'n1-standard-1', cpu: 1, ram: 3750 },
    MStandard2: { id: 'n1-standard-2', cpu: 2, ram: 7500 },
    MStandard4: { id: 'n1-standard-4', cpu: 4, ram: 15000 },
    MStandard8: { id: 'n1-standard-8', cpu: 8, ram: 30000 },
    MStandard16: { id: 'n1-standard-16', cpu: 16, ram: 60000 },
    MStandard32: { id: 'n1-standard-32', cpu: 32, ram: 120000 },
};
