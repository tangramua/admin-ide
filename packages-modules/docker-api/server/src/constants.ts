export enum Types {
    RegistryService = 'RegistryService',
    RegistryManagement = 'RegistryManagement',
    UniversalRegistryService = 'UniversalRegistryService',
}
