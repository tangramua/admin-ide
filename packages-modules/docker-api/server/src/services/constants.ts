export enum REGISTRY_ROUTES {
    TAGS = 'tags',
    SEARCH = 'search',
    REGISTRY = 'registry',
    CONTAINERS = 'containers',
}
