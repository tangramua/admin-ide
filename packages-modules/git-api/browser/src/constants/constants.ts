export const GIT_API_NAMESPACE = '@git-api';

export const ID_TOKEN = 'id_token';

// routes
export enum GIT_API_ROUTES {
    GIT_REPOSITORIES = '@git-api/repositories',
    GIT_REPOSITORY = '@git-api/repository',
}

