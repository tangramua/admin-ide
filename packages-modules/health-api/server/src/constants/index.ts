export * from './types';
export * from './hemera';
