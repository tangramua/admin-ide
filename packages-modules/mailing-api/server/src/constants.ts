export enum Types {
    MailerService = 'MailerService',
    TemplateRepository = 'TemplateRepository',
    MailReportRepository = 'MailReportRepository',
}
