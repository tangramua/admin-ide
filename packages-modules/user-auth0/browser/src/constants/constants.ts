export type AUTH0_NAMESPACE = '@auth0';
export const AUTH0_NAMESPACE: AUTH0_NAMESPACE = '@auth0';

export type CHECKING_OUT_START = '@auth0/CHECKING_OUT_START';
export const CHECKING_OUT_START: CHECKING_OUT_START = '@auth0/CHECKING_OUT_START';

export type CHECKING_OUT_END = '@auth0/CHECKING_OUT_END';
export const CHECKING_OUT_END: CHECKING_OUT_END = '@auth0/CHECKING_OUT_END';

export type PROFILE_FETCHED = '@auth0/PROFILE_FETCHED';
export const PROFILE_FETCHED: PROFILE_FETCHED = '@auth0/PROFILE_FETCHED';

export type PROFILE_FETCH_FAILED = '@auth0/PROFILE_FETCH_FAILED';
export const PROFILE_FETCH_FAILED: PROFILE_FETCH_FAILED = '@auth0/PROFILE_FETCH_FAILED';

export type PROFILE_FETCHING = '@auth0/PROFILE_FETCHING';
export const PROFILE_FETCHING: PROFILE_FETCHING = '@auth0/PROFILE_FETCHING';

export type LOGIN_SUCCESS = '@auth0/LOGIN_SUCCESS';
export const LOGIN_SUCCESS: LOGIN_SUCCESS = '@auth0/LOGIN_SUCCESS';

export type LOGGED_OUT = '@auth0/LOG_OUT';
export const LOGGED_OUT: LOGGED_OUT = '@auth0/LOG_OUT';

export type LOGIN_FAILED = '@auth0/LOGIN_FAILED';
export const LOGIN_FAILED: LOGIN_FAILED = '@auth0/LOGIN_FAILED';

export type AUTH0_LOGIN_SUCCESS = '@auth0/AUTH0_LOGIN_SUCCESS';
export const AUTH0_LOGIN_SUCCESS: AUTH0_LOGIN_SUCCESS = '@auth0/AUTH0_LOGIN_SUCCESS';

export type LOGGING_IN_TO_PROCEED = '@auth0/LOGGING_IN_TO_PROCEED';
export const LOGGING_IN_TO_PROCEED: LOGGING_IN_TO_PROCEED = '@auth0/LOGGING_IN_TO_PROCEED';

export type LOGGING_IN_PROGRESS = '@auth0/LOGGING_IN_PROGRESS';
export const LOGGING_IN_PROGRESS: LOGGING_IN_PROGRESS = '@auth0/LOGGING_IN_PROGRESS';

//Redirects
export type LOGIN_REDIRECT_PATH = '@auth0/LOGIN_REDIRECT_PATH';
export const LOGIN_REDIRECT_PATH: LOGIN_REDIRECT_PATH = '@auth0/LOGIN_REDIRECT_PATH';

export type LOGOUT_REDIRECT_PATH = '@auth0/LOGOUT_REDIRECT_PATH';
export const LOGOUT_REDIRECT_PATH: LOGOUT_REDIRECT_PATH = '@auth0/LOGOUT_REDIRECT_PATH';

export const ID_TOKEN = 'id_token';
export const ACCESS_TOKEN = 'access_token';


// Routes
export enum AUTH0_ROUTES {
    LOGIN = '@auth0/login',
    CALLBACK = '@auth0/callback',
    PROFILE = '@auth0/profile',
    LOGOUT = '@auth0/logout',
}


export const LOGIN_REDIRECT_STORAGE_PATH = 'login_redirect_path';
export const LOGOUT_REDIRECT_STORAGE_PATH = 'logout_redirect_path';

