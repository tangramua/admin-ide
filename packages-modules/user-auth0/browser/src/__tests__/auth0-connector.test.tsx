
import { Feature } from '@common-stack/client-react';
import { AUTH0_ROUTES } from '../constants';
import Auth0Module from '../index';
import 'jest';

describe('connector getRoutes', () => {
    test('with static routes and configurable routes', async () => {

        const { routeConfig, ...auth0Rest } = Auth0Module as any;
        const authRoutes = routeConfig[0];
        const login = authRoutes[AUTH0_ROUTES.LOGIN];
        const logout = authRoutes[AUTH0_ROUTES.LOGOUT];
        const callback = authRoutes[AUTH0_ROUTES.CALLBACK];
        const profile = authRoutes[AUTH0_ROUTES.PROFILE];
        const connector = new Feature({
            routeConfig: [{
                ['/login']: { ...login },
                ['/logout']: { ...logout },
                ['/callback']: { ...callback },
            }],
        }, Auth0Module);

        const routes = connector.getRoutes();

        expect(routes).toMatchSnapshot();
    });

});

