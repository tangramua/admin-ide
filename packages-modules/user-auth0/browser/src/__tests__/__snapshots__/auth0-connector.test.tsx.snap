// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connector getRoutes with static routes and configurable routes 1`] = `
<Switch>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={true}
    path="/logout"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={true}
    path="/callback"
    render={[Function]}
    strict={undefined}
  />
</Switch>
`;
